services/checks/checks.go

```diff
@@ -172,8 +177,44 @@ func (s *Service) Run(ctx context.Context) {
 		return
 	}
 
-	ticker := time.NewTicker(s.interval)
-	defer ticker.Stop()
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		s.resendAlerts(ctx)
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		s.restartChecks(ctx)
+	}()
+
+	wg.Wait()
+}
+
+// resendAlerts resends collected alerts until ctx is canceled.
+func (s *Service) resendAlerts(ctx context.Context) {
+	t := time.NewTicker(s.alertsRegistry.resendInterval)
```

Comment:
That (and TTL) should be a property of the service, not registry

Reply:
I did that in the latest push but for registry unit tests I can no longer use `resendInterval` and had to use the constant `defaultResendInterval`